modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/NameClashModelValidator.java

```diff
@@ -213,6 +258,68 @@ private void validateClash(String existingNamingModel, String newNamingModel, St
                                                          typeOfNewNamingModel));
       }
     }
+
+    private void validateSingularizingClash(ParameterizedModel model) {
+      List<ParameterModel> parameters = model.getParameterModels();
+      parameters.forEach(
+                         p -> p.getType().accept(new MetadataTypeVisitor() {
```

Comment:
create the visitor once, maybe even using a method to be more clear on what it is for. This way we avoid creating it N times, and also simplify this scope